src/Tizen.Sensor/Tizen.Sensor/Sensor.cs

```diff
@@ -545,5 +638,623 @@ private void DestroyHandles()
+            }
+
+        }
+
+        /// <summary>
+        /// Registers the callback function to be invoked when a sensor listener stops the sensor provider.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void SetProviderStopCb()
+        {
+            _sensorProviderStopCallBack = (IntPtr _provider, Int64 _userData) =>
+            {
+            };
+
+            int error = Interop.SensorProvider.SetProviderStopCb(_sensorProviderHandle, _sensorProviderStopCallBack, IntPtr.Zero);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in setting stop callback of sensor provider");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetStopCb Failed");
+            }
+
+        }
+
+        /// <summary>
+        /// Registers the callback function to be invoked when the interval is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void SetProviderIntervalChangedCb()
+        {
+            _sensorProviderIntervalChangedCallBack = (IntPtr _provider, uint IntervalMs, Int64 _userData) =>
+            {
+                ProviderIntervalChangedEventArgs e = new ProviderIntervalChangedEventArgs(_provider, IntervalMs, _userData)
+                {
+                };
+                ProviderIntervalChanged?.Invoke(null, e);
+            };
+
+            int error = Interop.SensorProvider.SetProviderIntervalChangedCb(_sensorProviderHandle, _sensorProviderIntervalChangedCallBack, IntPtr.Zero);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in setting interval callback of sensor provider");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetIntervalChangedCb Failed");
+            }
+
+        }
+
+        /// <summary>
+        /// Publishes a sensor event through the declared sensor.
+        /// This function publishes a sensor's data to its listeners.
+        /// </summary>
+        /// <param name="sensorEventS">
+        /// The sensor event.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void ProviderPublish(SensorEventStruct sensorEventS)
+        {
+            int error = Interop.SensorProvider.ProviderPublish(_sensorProviderHandle, sensorEventS);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in sensor provider publish");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderPublish Failed");
+            }
+
+        }
+
+        /// <summary>
+        /// Adds a callback function to be invoked when a new sensor is added.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void SetSensorAddedCB()
```

Comment:
1. Do not use callback, use event
2. API name is `Set` but can't pass a value to set